src/modules/SimpleSlippageRouter.sol

```diff
@@ -0,0 +1,132 @@
+        _cellar.deposit(_assets, msg.sender);
+    }
+
+    /**
+     * @notice withdraws assets as long as tx returns more than _assets and is done before _deadline.
+     * @param _cellar specified cellar to withdraw assets from.
+     * @param _assets amount of cellar base assets to withdraw.
+     * @param _maxShares max amount of shares to redeem from tx.
+     * @param _deadline block.timestamp that tx must be carried out by.
+     */
+    function withdraw(Cellar _cellar, uint256 _assets, uint256 _maxShares, uint64 _deadline) public {
+        if (block.timestamp > _deadline) revert SimpleSlippageAdaptor__ExpiredDeadline(_deadline);
+
+        uint256 shares = _cellar.previewWithdraw(_assets);
+        if (shares > _maxShares) revert SimpleSlippageAdaptor__WithdrawMaxSharesSurpassed(_maxShares, shares);
+
+        _cellar.withdraw(_assets, msg.sender, msg.sender); // NOTE: user needs to approve this contract to send shares
+    }
+
+    /**
+     * @notice mints shares from the cellar and returns shares to receiver IF shares quoted cost are less than specified _assets amount by the specified _deadline.
+     * @param _cellar specified cellar to deposit assets into.
+     * @param _maxAssets max amount of cellar base assets to deposit.
+     * @param _minimumShares amount of shares required at min from tx.
+     * @param _deadline block.timestamp that tx must be carried out by.
+     */
+    function mint(Cellar _cellar, uint256 _maxAssets, uint256 _minimumShares, uint64 _deadline) public {
+        if (block.timestamp > _deadline) revert SimpleSlippageAdaptor__ExpiredDeadline(_deadline);
+        uint256 quotedAssetAmount = _cellar.previewMint(_minimumShares);
+        if (quotedAssetAmount > _maxAssets)
+            revert SimpleSlippageAdaptor__MintMinimumSharesSurpassed(_minimumShares, _maxAssets, quotedAssetAmount);
+        _cellar.asset().safeTransferFrom(msg.sender, address(this), quotedAssetAmount);
+        _cellar.asset().approve(address(_cellar), quotedAssetAmount);
+        _cellar.mint(_minimumShares, msg.sender);
+    }
+
+    /**
+     * @notice redeem shares to withdraw assets from the cellar IF withdrawn quotedAssetAmount > _minAssets & tx carried out before _deadline.
+     * @param _cellar specified cellar to redeem shares for assets from.
+     * @param _minAssets amount of cellar base assets to receive upon share redemption.
+     * @param _maxShares max amount of shares to redeem from tx.
+     * @param _deadline block.timestamp that tx must be carried out by.
+     */
+    function redeem(Cellar _cellar, uint256 _minAssets, uint256 _maxShares, uint64 _deadline) public {
+        if (block.timestamp > _deadline) revert SimpleSlippageAdaptor__ExpiredDeadline(_deadline);
+
+        uint256 quotedAssetAmount = _cellar.previewRedeem(_maxShares);
+        if (quotedAssetAmount < _minAssets)
+            revert SimpleSlippageAdaptor__RedeemMaxSharesSurpassed(_maxShares, _minAssets, quotedAssetAmount);
```

Comment:
This error should be more like minimum assets not met

Reply:
Good point, renamed it